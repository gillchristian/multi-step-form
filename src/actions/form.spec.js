import {
  submitForm,
  submitFormEnd,
  validateAsyncBegin,
  validateAsyncEnd,
  moveToNextStep,
  onFieldChange,
} from './form'

describe('action creators', () => {

  describe('submitForm', () => {
    it('creates a SUBMIT_FORM_BEGIN action', () => {
      const action = submitForm()

      expect(action).toMatchSnapshot()
    })
  })

  describe('submitFormEnd', () => {
    it('creates a SUBMIT_FORM_END action with error', () => {
      const action = submitFormEnd(new Error('some error'))

      expect(action).toMatchSnapshot()
    })

    it('creates a SUBMIT_FORM_END action with no error', () => {
      const action = submitFormEnd()

      expect(action).toMatchSnapshot()
    })
  })

  describe('validateAsyncBegin', () => {
    it('creates an action with the provided field as meta', () => {
      const action = validateAsyncBegin('field-name')()

      expect(action).toMatchSnapshot()
    })
  })

  describe('validateAsyncEnd', () => {
    it('creates an action with the provided field as meta and payload', () => {
      const action = validateAsyncEnd('field-name')({foo: 'bar'})

      expect(action).toMatchSnapshot()
    })

    it('creates an action with the provided field as meta and error', () => {
      const action = validateAsyncEnd('field-name')(new Error('some error'))

      expect(action).toMatchSnapshot()
    })
  })

  describe('moveToNextStep', () => {
    it('creates a MOVE_TO_NEXT_STEP action', () => {
      const action = moveToNextStep()

      expect(action).toMatchSnapshot()
    })
  })

  describe('onFieldChange', () => {
    it('creates an ON_FIELD_CHANGE action with meta and payload', () => {
      const action = onFieldChange('field-name', 'some value')

      expect(action).toMatchSnapshot()
    })
  })
})

