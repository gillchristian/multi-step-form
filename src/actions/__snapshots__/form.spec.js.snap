exports[`action creators moveToNextStep creates a MOVE_TO_NEXT_STEP action 1`] = `
Object {
  "type": "form:steps:move-to-next",
}
`;

exports[`action creators onFieldChange creates an ON_FIELD_CHANGE action with meta and payload 1`] = `
Object {
  "meta": "field-name",
  "payload": Object {
    "value": "some value",
  },
  "type": "form:data:update-field",
}
`;

exports[`action creators submitForm creates a SUBMIT_FORM_BEGIN action 1`] = `
Object {
  "type": "form:submition:begin",
}
`;

exports[`action creators submitFormEnd creates a SUBMIT_FORM_END action with error 1`] = `
Object {
  "error": true,
  "payload": [Error: some error],
  "type": "form:submition:end",
}
`;

exports[`action creators submitFormEnd creates a SUBMIT_FORM_END action with no error 1`] = `
Object {
  "type": "form:submition:end",
}
`;

exports[`action creators validateAsyncBegin creates an action with the provided field as meta 1`] = `
Object {
  "meta": "field-name",
  "type": "form:validation:async-begin",
}
`;

exports[`action creators validateAsyncEnd creates an action with the provided field as meta and error 1`] = `
Object {
  "error": true,
  "meta": "field-name",
  "payload": [Error: some error],
  "type": "form:validation:async-end",
}
`;

exports[`action creators validateAsyncEnd creates an action with the provided field as meta and payload 1`] = `
Object {
  "meta": "field-name",
  "payload": Object {
    "foo": "bar",
  },
  "type": "form:validation:async-end",
}
`;
